### PR TITLE
Increased hostname buffer size, replaced sasl with simple bind

### PR DESCRIPTION
**Pull request by pythianrenaud**

### CODE REVIEW

File: library/addisco.c [hunk 1]
```diff
@@ -155,7 +155,7 @@ get_string (unsigned char *beg,
             unsigned char *end,
             unsigned char **at)
 {
-	char buffer[HOST_NAME_MAX];
+	char buffer[75];
```
Comment (tehamachartrand) [inline library/addisco.c:160]: why 75 ?

**Review by tehamachartrand — APPROVED** [review]

File: library/addisco.c (hunk 1 above)
Comment (pythianrenaud) [reply library/addisco.c:160]: chose it arbitrarily, gives us enough space for our max room length

Comment (varxint) [reply library/addisco.c:160]: Might be worth a comment to explain this for the person who is supporting this in late 2021 and beyond.

File: library/adconn.c [hunk 2]
```diff
@@ -1094,10 +1097,21 @@ authenticate_to_directory (adcli_conn *conn)
...
+	/* Populate the userDn string based on the user and naming context */
+	strncpy(username, conn->user_name, sizeof(username) - 1);
+	username[sizeof(username) - 1] = '\0';
+	for (int i = 0; i < strlen(username); i++) {
+		if (username[i] == '@') {
+			username[i] = '\0';
+			break;
+		}
+	}
+	snprintf(userDn, sizeof(userDn), "CN=%s,CN=Users,%s", username, conn->default_naming_context);
```
Comment (varxint) [inline library/adconn.c:1110]: It's assured that conn->default_naming_context will never exceed 79 characters?

Comment (pythianrenaud) [reply library/adconn.c:1110]: correct, at most it can be 63 characters

File: library/addisco.c (hunk 1 above)
Comment (pythianrenaud) [reply library/addisco.c:160]: sure thing

**Pull request merged**